plane_spotter/twitter.py

```diff
@@ -98,11 +101,16 @@ def login(self, log: structlog.stdlib.BoundLogger = logger):
 
         self.unusual_activity_challenge(log=log)
 
+        password_xpath_1 = "/html/body/div/div/div/div[1]/div/div/div/div/div/div/div[2]/div[2]/div/div/div[2]/div[2]/div[1]/div/div/div[3]/div/label/div/div[2]/div[1]/input"
+        password_xpath_2 = "/html/body/div[1]/div/div/div[1]/div/div/div/div/div/div/div[2]/div[2]/div/div/div[2]/div[2]/div[1]/div/div/div[3]/div/label/div/div[2]/div[1]/input"
+
+        log.info("getting password input")
+
+        password = self._get_password([password_xpath_1, password_xpath_2])
+        if password is None:
+            raise RuntimeError("Failed to get password element, check xpaths")
```

Comment:
I think it may actually be easier to find the element by doing something like:

```python
self._wait_until_clickable(By.CSS_SELECTOR, 'whatever the selector is')
```

The XPATH method is fragile, could you see if you could make that work?

Reply:
the problem i have with css selectors is that they (at least imo) seem much more likely to change versus an xpath -- e.g. UI devs purposefully altering the already obfuscated class names

Although I don't have a _ton_ of selenium experience, it just seemed to me back when i worked with it in node that xpath selectors were actually more reliable.

can definitely get a css selector to work though, just not sure how long it'll continue working lol

Reply:
Hmm, you may be right. The element is:

```
<input autocapitalize="sentences" autocomplete="username" autocorrect="on" name="text" spellcheck="true" type="text" dir="auto" class="r-30o5oe r-1niwhzg r-17gur6a r-1yadl64 r-deolkf r-homxoj r-poiln3 r-7cikom r-1ny4l3l r-t60dpp r-1dz5y72 r-fdjqy7 r-13qz1uu" value="">
```

There's not anything particularly useful or identifying there so maybe xpath is the only option. Oh well.